### PR TITLE
IPC refactor 

### PR DESCRIPTION
**Pull request by kewde**
Refactored the IPC service

- [x] you can now register listeners on the renderer side
- [x] added testcase/example for ZMQ integration
- [x] move code, streamline ipc a bit more
- [x] moved notifications

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -1,26 +1,27 @@
 
-const electron      = require('electron');
```
Comment (kewde) [inline main.js]: aww, my formatter killed this.. Can someone check this out?

**Comment by kewde** [comment]
The karma tests are giving an error due to the fact that the IPC is not available in the Chromium browser.
We can look into running karma with electron however:
https://github.com/twolfson/karma-electron

Or I can make a small "mock" test to inject, which will emulate the IPC and get rid of the errors.

File: main.js (hunk 1 above)
Comment (pciavald) [reply main.js]: what do you mean @kewde ?

**Comment by pciavald** [comment]
@kewde won't merge this in dev right now, right ?

**Comment by kewde** [comment]
@pciavald nope, it's for the 1.1.1

File: main.js (hunk 1 above)
Comment (kewde) [reply main.js]: @pciavald I liked the previous formatting, where the '=' are all aligned vertically.

**Pull request merged**